File: frontend/src/review/object/EligibilityReviewRequest.ts
import type {
  EligibilityReviewTarget,
  EntityId,
  ReasonText,
} from '@/shared/object/domain/DomainObjects'

export type EligibilityReviewRequest = {
  target: EligibilityReviewTarget
  targetId: EntityId
  reason: ReasonText
}

File: backend/src/main/scala/review/object/EligibilityReviewRequest.scala
package domain.review

import domain.shared.given

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*
import domain.shared.*

final case class EligibilityReviewRequest(
    target: EligibilityReviewTarget,
    targetId: EntityId,
    reason: ReasonText,
)
object EligibilityReviewRequest:
  given Encoder[EligibilityReviewRequest] = deriveEncoder
  given Decoder[EligibilityReviewRequest] = deriveDecoder
